### PR TITLE
Add json-path input support for enrich mediator.

### PR DESCRIPTION
**Pull request by GDLMadushanka**

### CODE REVIEW

File: components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/java/org/wso2/carbon/mediator/enrich/ui/EnrichMediator.java [hunk 1]
```diff
@@ -21,8 +21,7 @@
 import org.apache.axiom.om.OMElement;
 import org.apache.axiom.om.impl.builder.StAXOMBuilder;
 import org.apache.axiom.om.util.AXIOMUtil;
-import org.apache.synapse.config.xml.SynapseXPathFactory;
-import org.apache.synapse.config.xml.SynapseXPathSerializer;
+import org.apache.synapse.config.xml.*;
```
Comment (madushadhanushka) [inline components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/java/org/wso2/carbon/mediator/enrich/ui/EnrichMediator.java]: Wild card import

Comment (GDLMadushanka) [reply components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/java/org/wso2/carbon/mediator/enrich/ui/EnrichMediator.java]: +1 fixed

File: components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/resources/web/enrich-mediator/edit-mediator.jsp [hunk 2]
```diff
@@ -89,11 +92,13 @@
...
-            SynapseXPath targetXPath = enrichMediator.getTargetExpression();
-            NameSpacesRegistrar nameSpacesRegistrar = NameSpacesRegistrar.getInstance();
-            if (targetXPath != null) {
-                nameSpacesRegistrar.registerNameSpaces(targetXPath, "mediator.enrich.target.val_ex", session);
-                targetValue = targetXPath.toString();
+            SynapsePath synapsePath = enrichMediator.getTargetExpression();
+            if (synapsePath != null) {
+                if (synapsePath instanceof SynapseXPath) {
+                    NameSpacesRegistrar nameSpacesRegistrar = NameSpacesRegistrar.getInstance();
+                    nameSpacesRegistrar.registerNameSpaces(synapsePath, "mediator.enrich.target.val_ex", session);
```
Comment (a5anka) [inline components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/resources/web/enrich-mediator/edit-mediator.jsp]: Duplicate code in line 65-70.

File: components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/resources/web/enrich-mediator/update-mediator.jsp [hunk 3]
```diff
@@ -35,12 +37,23 @@
     if (request.getParameter("mediator.enrich.source.val_ex") != null && !request.getParameter("mediator.enrich.source.val_ex").trim().equals("")) {
 
         if (request.getParameter("mediator.enrich.source.type2").equals("custom")) {
-            XPathFactory xPathFactory = XPathFactory.getInstance();
-            enrichMediator.setSourceExpression(xPathFactory.createSynapseXPath("mediator.enrich.source.val_ex",
-                    request.getParameter("mediator.enrich.source.val_ex"), session));
+            String expression = request.getParameter("mediator.enrich.source.val_ex");
+            if (expression.startsWith("json-eval(")) {
+                int expLength = expression.length();
+                String extractedExp = expression.substring(10, expLength - 1);
```
Comment (a5anka) [inline components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/resources/web/enrich-mediator/update-mediator.jsp]: We have a magic number here. Shall we fix?

File: components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/resources/web/enrich-mediator/update-mediator.jsp [hunk 4]
```diff
@@ -67,11 +80,22 @@
 
     if (request.getParameter("mediator.enrich.target.val_ex") != null && !request.getParameter("mediator.enrich.target.val_ex").trim().equals("")) {
         if (request.getParameter("mediator.enrich.target.type").equals("custom")) {
-            XPathFactory xPathFactory2 = XPathFactory.getInstance();
-            enrichMediator.setTargetExpression(xPathFactory2.createSynapseXPath("mediator.enrich.target.val_ex",
-                    request.getParameter("mediator.enrich.target.val_ex"),session));
+            String expression = request.getParameter("mediator.enrich.target.val_ex");
```
Comment (a5anka) [inline components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/resources/web/enrich-mediator/update-mediator.jsp]: Duplicate code in 40-52.

**Pull request closed without merging**